ee/clickhouse/queries/paths/path_event_query.py

```diff
@@ -35,16 +23,30 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
 
         _fields = [
             f"{self.EVENT_TABLE_ALIAS}.timestamp AS timestamp",
-            (
-                f"if(event = '{SCREEN_EVENT}', {self._get_screen_name_parsing()}, "
-                f"if({self.EVENT_TABLE_ALIAS}.event = '{PAGEVIEW_EVENT}', {self._get_current_url_parsing()}, "
-                f"if({self.EVENT_TABLE_ALIAS}.event = '{AUTOCAPTURE_EVENT}', concat('autocapture:', {self.EVENT_TABLE_ALIAS}.elements_chain), "
-                f"{self.EVENT_TABLE_ALIAS}.event))) AS path_item"
-            ),
             f"{self.DISTINCT_ID_TABLE_ALIAS}.person_id as person_id" if self._should_join_distinct_ids else "",
             funnel_paths_timestamp,
         ]
 
+        event_conditional = (
+            f"if({self.EVENT_TABLE_ALIAS}.event = '{SCREEN_EVENT}', {self._get_screen_name_parsing()}, "
+            if self._column_optimizer.should_query_screen_in_paths
+            else "if(0, '', "
+        )
+        event_conditional += (
+            f"if({self.EVENT_TABLE_ALIAS}.event = '{PAGEVIEW_EVENT}', {self._get_current_url_parsing()}, "
+            if self._column_optimizer.should_query_url_in_paths
```

Comment:
This is really abusing column_optimizer and violating it's one goal.

ColumnOptimizer is - given a filter - responsible for figuring out what _columns_ one needs to query to make it happen. It should ideally contain _no_ business logic (though there's a bit involved in figuring out what properties are even used).

Here it's being used as a paths business logic container and used for figuring out deep filtering logic (on events still?). We should not extend column_optimizer this way.

The reason we're in this mess is mismatching abstractions - every other query puts things it wants to query under `filter.entities`. 

Consider doing the same here and keeping the "hey, what does the user wants" bits under the `PathsFilter` object - either directly or creating paths-specific helper functions that take it as an argument.